pkg/util/policy/agent/agent.py

```diff
@@ -284,138 +258,231 @@ def make_new_full_ruleset(current_rules, new_rules):
+        tenant_wide_policy_vector_chain = "ROMANA-T%s-W" % tenant
+
+        # The name for the new policy's chain(s). Need to include the tenant ID to
+        # avoid name conflicts.
+        policy_chain_name = "ROMANA-T%dP-%s_" % \
+            (tenant, name)
+
+        # Policy chain only hosts match conditions, rules themselves are
+        # applied in this auxiliary chain
+        in_chain_name  = policy_chain_name[:-1] + "-IN_"
+
+        # Per segment policy chain to host jumps to the actuall policy chains.
+        if target_segment:
+            target_segment_forward_chain = "ROMANA-T%s-S%s" % \
+                (tenant, target_segment)
+        else:
+            # If segment_network_id not provided in the policy,
+            # consider policy to be tenant wide.
+            target_segment_forward_chain = tenant_wide_policy_vector_chain
+
+        # Chain names are going to be used later to fill in the rules. Store them.
+        policy_chains[in_chain_name] = True
+
+        # Jump from per-tenant chain into per-segment chains and default DROP.
+        rules[tenant_policy_vector_chain] = [
+            _make_rule(tenant_policy_vector_chain, "-j %s" % target_segment_forward_chain),
+            _make_rule(tenant_policy_vector_chain, "-j %s" % tenant_wide_policy_vector_chain),
+            _make_rule(tenant_policy_vector_chain, "-j DROP")
+        ]
+
+        # Jump from per-segment chain into policy chain
+        if target_segment_forward_chain not in rules:
+            rules[target_segment_forward_chain] = []
+            rules[target_segment_forward_chain].append(
+                    _make_rule(target_segment_forward_chain, '-m comment --comment POLICY_CHAIN_HEADER -j RETURN'))
+        rules[target_segment_forward_chain].insert(0,
+                _make_rule(target_segment_forward_chain, "-j %s" % policy_chain_name))
+
+        # Default rules per tenant.
+        if tenant_wide_policy_vector_chain not in rules:
+            rules[tenant_wide_policy_vector_chain] = []
+            rules[tenant_wide_policy_vector_chain].append(
+                _make_rule(tenant_wide_policy_vector_chain, '-m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT'))
+            rules[tenant_wide_policy_vector_chain].append(
+                _make_rule(tenant_wide_policy_vector_chain, '-m comment --comment POLICY_CHAIN_HEADER -j RETURN'))
+
+        # Loop over peers and fill top level policy chains with source matching rules
+        for peer in policy_def.get('peers'):
+            # Possible peers are
+            # CIDR - detected by cidr field
+            # peer - detected by peer type
+            # tid/sid - detected by tenant_network_id and segment_network_id
+            pr = peer.get('peer')
+            cidr = peer.get('cidr')
+            from_tenant = peer.get('tenant_network_id')
+            from_segment = peer.get('segment_network_id')
+
+            if pr:
+                if pr == "any":
```

Comment:
yea, on second thought, you're right - boolean is not obvious (ingress/egress is also a binary value but it's obvious when it's one of the two). In this case it could be *all: true* but let's keep it the way you have already.